internal/commands/user/revoke.go

```diff
@@ -0,0 +1,151 @@
+	cmd.realmClient = profile.RealmAuthClient()
+	return nil
+}
+
+// Handler is the command handler
+func (cmd *CommandRevoke) Handler(profile *cli.Profile, ui terminal.UI) error {
+	app, appErr := app.Resolve(ui, cmd.realmClient, cmd.inputs.Filter())
+	if appErr != nil {
+		return appErr
+	}
+	users, usersErr := cmd.inputs.ResolveUsers(ui, cmd.realmClient, app)
+	if usersErr != nil {
+		return usersErr
+	}
+	for _, user := range users {
+		err := cmd.realmClient.RevokeUserSessions(app.GroupID, app.ID, user.ID)
+		cmd.outputs = append(cmd.outputs, userOutput{user: user, err: err})
+	}
+	return nil
+}
+
+// Feedback is the command feedback
+func (cmd *CommandRevoke) Feedback(profile *cli.Profile, ui terminal.UI) error {
+	if len(cmd.outputs) == 0 {
+		return ui.Print(terminal.NewTextLog("No users to revoke, try changing the --user input"))
+	}
+	var outputsByProviderType = map[realm.AuthProviderType][]userOutput{}
+	for _, output := range cmd.outputs {
+		for _, identity := range output.user.Identities {
+			outputsByProviderType[identity.ProviderType] = append(outputsByProviderType[identity.ProviderType], output)
+		}
+	}
+	logs := make([]terminal.Log, 0, len(outputsByProviderType))
+	for _, apt := range realm.ValidAuthProviderTypes {
+		outputs := outputsByProviderType[apt]
+		if len(outputs) == 0 {
+			continue
+		}
+		sort.SliceStable(outputs, getUserOutputComparerBySuccess(outputs))
+		logs = append(logs, terminal.NewTableLog(
+			fmt.Sprintf("Provider type: %s", apt.Display()),
+			userRevokeTableHeaders(apt),
+			userRevokeTableRows(apt, outputs)...,
+		))
+	}
+	return ui.Print(logs...)
+}
+
+func userRevokeTableHeaders(authProviderType realm.AuthProviderType) []string {
+	var headers []string
+	switch authProviderType {
+	case realm.AuthProviderTypeAPIKey:
+		headers = append(headers, headerName)
+	case realm.AuthProviderTypeUserPassword:
+		headers = append(headers, headerEmail)
+	}
+	headers = append(
+		headers,
+		headerID,
+		headerType,
+		headerRevoked,
+		headerDetails,
+	)
+	return headers
+}
+
+func userRevokeTableRows(authProviderType realm.AuthProviderType, outputs []userOutput) []map[string]interface{} {
+	rows := make([]map[string]interface{}, 0, len(outputs))
+	for _, output := range outputs {
+		rows = append(rows, userRevokeTableRow(authProviderType, output))
+	}
+	return rows
+}
+
+func userRevokeTableRow(authProviderType realm.AuthProviderType, output userOutput) map[string]interface{} {
+	var details string
+	if output.err != nil {
+		details = output.err.Error()
+	}
+	success := "no"
+	if output.err == nil {
+		success = "yes"
+	}
+	row := map[string]interface{}{
+		headerID:      output.user.ID,
+		headerType:    output.user.Type,
+		headerRevoked: success,
+		headerDetails: details,
+	}
+	switch authProviderType {
+	case realm.AuthProviderTypeAPIKey:
+		row[headerName] = output.user.Data[userDataName]
+	case realm.AuthProviderTypeUserPassword:
+		row[headerEmail] = output.user.Data[userDataEmail]
+	}
+	return row
+}
```

Comment:
I pretty much just took this from `user delete`. I don't mind refactoring and reusing here though I actually don't mind keeping the code in the commands separate in this case. Any strong preferences for reusing?

Reply:
I think we should definitely start to refactor these feedback helper functions since there is just tons of lines that are identical.

Reply:
Yeah, the super simple way would be to change `headerRevoked` to something generic like `Successful` and then we could reuse the code across commands  pretty easily. Not sure if we have a standard on this or not.

Reply:
my thinking here is that a lot of this code isn't necessarily a bad thing ... it's explicit about what it does for each command.  that being said, we can definitely spruce it up a bit and reduce a bit of the repetition here.

i think its about time we start with some kind of `TableBuilder` concept, to hopefully alleviate some of the pains and awkward code here.  It's could just construct the `[]string` and `[]map[string]interface{}` pairs for printing tables.  We can chat about this offline in more detail